creator-node/src/config.js

```diff
@@ -414,6 +414,19 @@ const config = convict({
     format: 'nat',
     env: 'maxStorageUsedPercent',
     default: 95
+  },
+
+  pinAddCIDs: {
+    doc: 'Array of comma separated CIDs to pin',
+    format: String,
+    env: 'pinAddCIDs',
+    default: ''
+  },
+  pinRemoveCIDs: {
```

Comment:
is there a way to just check what is already pinned and make it the same as pinAddCIDs on each push? so we don't have to maintain two sep arrays

Reply:
@hareeshnagaraj 
Thoughts on running something like `ipfs pin ls --type recursive` and removing all old pins not in the set before adding the new ones? I know you said it was somewhat expensive.

Reply:
yeah it used to be quite expensive but given how we add files now (no pin op) it should be relatively fast - however it will be a bit weird if there are inner/indirect CIDs that get pinned as part of this dir add, not sure how to solve that problem. basically i'm saying if we add a CID `QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB` it's not 100% that this is the _only_ CID that gets added as a result

Reply:
i do think we should call pin add with recursive true if we don't

Reply:
looks like we do which is great

Reply:
ok ignore my earlier comments I understand - @jowlee  yes I think this is a great idea and much simpler we can explicitly check our 'recursive' direct pins and remove anything that doesn't match this CID. since we aren't pinning literally every single thing like before it shouldn't be a significant perf hit